File: example/LINUX/OMP/pi.c
#include <stdio.h>
#include <omp.h>
#include <math.h>

int main(int argc, char **argv)
{
	int i;
	int n = 1000000;
	double PI25DT = 3.141592653589793238462643;
	double pi, h, area, x, start, end;

	OMPtrace_init();

	start = omp_get_wtime();
	h = 1.0 / (double) n;
	area = 0.0;
	#pragma omp parallel for private(x) reduction(+:area)
	for (i = 1; i <= n; i++)
	{
		x = h * ((double)i - 0.5);
		area += (4.0 / (1.0 + x*x));
	}
	pi = h * area;
	end = omp_get_wtime();

	printf("pi is approximately %.16f, Error is %.16f\n",pi,fabs(pi - PI25DT));
	printf("Ran in %0.5f seconds with %d threads\n",end-start,omp_get_max_threads());

	OMPtrace_fini();
}

File: example/LINUX/OMP/pi.f
      PROGRAM pi

      integer i
      real x, sum, h

      h = 1.0 / 10000000

!$OMP PARALLEL PRIVATE(X, i)
!$OMP DO REDUCTION (+:sum)
      DO i = 1, 10000000, 1
       x = h*(i-0.5)
       sum = sum + 4.0/(1.0+x**2)
      ENDDO
!$OMP END DO 
!$OMP END PARALLEL


!$OMP PARALLEL PRIVATE(X, i)
!$OMP DO REDUCTION (+:sum)
      DO i = 1, 10000000, 1
       x = h*(i-0.5)
       sum = sum + 4.0/(1.0+x**2)
      ENDDO
!$OMP END DO 
!$OMP END PARALLEL

      write (*,*) "pi = ",sum*h
 
      stop

      END
